File: exercism/python/raindrops/raindrops.py
# list of tuples: (factor, sound)
RULES = [(3, 'Pling'), (5, 'Plang'), (7, 'Plong')]


def convert(n):
    sounds = [sound
              for (factor, sound) in RULES
              if n % factor == 0]
    return ''.join(sounds) if sounds else str(n)

File: exercism/clojure/raindrops/src/raindrops.clj
(ns raindrops
  (:require [clojure.string :as str]))

(defn div?
  "is n divisible by div?"
  [n div]
  (zero? (rem n div)))

(defn convert [n]
  (let [s (str/join [(if (div? n 3) "Pling" "")
                     (if (div? n 5) "Plang" "")
                     (if (div? n 7) "Plong" "")])]
    (if (= s "") (str n) s)))

